test/unit/os/env_spec.lua

```diff
@@ -127,4 +129,66 @@ describe('env function', function()
       eq(hostname, (ffi.string(hostname_buf)))
     end)
   end)
+
+  function vim_setenv(name, value)
+    return env.vim_setenv((to_cstr(name)), (to_cstr(value)))
+  end
+
+  describe('vim_setenv', function()
+    local env_nvimruntime
+    local env_nvim
+
+    setup(function()
+      -- Save vars or '', nil is not acceptable and there is no unsetenv
```

Comment:
related, though we don't need to worry about it for now: https://groups.google.com/d/msg/vim_dev/IIrDqVvdpOo/Q2nuImgZEtEJ